js/src/main/scala/crystal/react/reuse/Reuse.scala

```diff
@@ -33,22 +33,26 @@ import scala.reflect.ClassTag
 trait Reuse[+A] {
   type B
 
-  val get: () => A
+  lazy val value: A = getValue()
 
-  lazy val value: A = get()
+  protected[reuse] val getValue: () => A
 
   protected[reuse] val reuseBy: B // We need to store it to combine into tuples when currying.
 
   protected[reuse] implicit val ClassTag: ClassTag[B]
 
   protected[reuse] implicit val reusability: Reusability[B]
 
+  def addReuseBy[R: Reusability](r: R): Reuse[A] = Reuse.by((reuseBy, r))(value)
+  def addReuse[C](r: Reuse[C]): Reuse[A]         = addReuseBy(r.reuseBy)(r.reusability)
+
+  def replaceReuseBy[R: Reusability: ClassTag](r: R): Reuse[A] = Reuse.by(r)(value)
+  def replaceReuse[C](r: Reuse[C]): Reuse[A]                   = replaceReuseBy(r.reuseBy)(r.reusability, r.ClassTag)
+
```

Comment:
I have local changes that may reformulate these. I'll merge and send another PR shortly.